lstchain/reco/dl1_to_dl2.py

```diff
@@ -308,11 +308,11 @@ def build_models(filegammas, fileprotons,
         df_gamma = pd.concat([df_gamma, pd.read_hdf(filegammas, key=dl1_params_src_dep_lstcam_key)], axis=1)
         df_proton = pd.concat([df_proton, pd.read_hdf(fileprotons, key=dl1_params_src_dep_lstcam_key)], axis=1)
 
-    df_gamma = utils.filter_events(df_gamma, filters=events_filters)
-    df_proton = utils.filter_events(df_proton, filters=events_filters)
-
     regression_features = config['regression_features']
 
+    df_gamma = utils.filter_events(df_gamma, regression_features, filters=events_filters)
```

Comment:
rather than modifying `filter_events` that could be used in another context, you may simply:
```
regression_features = config['regression_features']
classification_features = config['classification_features']
all_features = list(set(regression_features + classification_features))

df_gamma = utils.filter_events(df_gamma[all_features], filters=events_filters)
df_proton = utils.filter_events(df_proton[all_features], filters=events_filters)
```

Reply:
Yeah, much better. This will only leave those `all_features` parameters though. Is that a problem for the reconstruction or only when filtering the events for applying the models as you mention below?

Reply:
although we cannot use `classification_features` for dl1 data which include `['log_reco_energy', 'reco_disp_dx', 'reco_disp_dy']`

Reply:
> Yeah, much better. This will only leave those `all_features` parameters though. Is that a problem for the reconstruction or only when filtering the events for applying the models as you mention below?

for training, in `build_models` it's fine since we load these events from files and don't use them any further

Reply:
> although we cannot use `classification_features` for dl1 data which include `['log_reco_energy', 'reco_disp_dx', 'reco_disp_dy']`

ha yes you are right, so instead of 
`all_features = list(set(regression_features + classification_features))`
`all_features = [f for f in set(regression_features + classification_features) if f in df_gamma.columns and f in df_proton.columns]`

Reply:
I think what we need is that filter_events returns all columns for rows in which the filters are fulfilled, and in which none of the regression features is nan. We could simply add that condition as additional filters, i.e. "xxx is not nan" (with xxx being each of the regression features),  and implement that in filter_events instead of the "dropna" call.